content/testing.md

```diff
@@ -0,0 +1,683 @@
+
+  - Using another package from the example app to isolate a publication, the `publication-collector` package:
+
+    ```js
+    describe('lists.public', () => {
+      it('sends all public lists', (done) => {
+        // Allows us to look at the output of a publication without needing a client connection
+        const collector = new PublicationCollector();
+        collector.collect('lists.public', (collections) => {
+          chai.assert.equal(collections.Lists.length, 3);
+          done();
+        });
+      });
+    });
+    ```
+
+There's a lot of scope for better isolation and testing utilities (the two packages from the example app above could be improved greatly!). We encourage the community to take the lead on these!
+
+<h2 id="integration-testing">Integration testing</h2>
+
+An integration test is a test that crosses module boundaries. In the simplest case, this simply means something very similar to a unit test, where you perform your isolation around multiple modules, creating a non-singular "system under test". 
+
+Although conceptually different to unit tests, such tests typically do not need to be run any differently to unit tests and can use the same [`meteor test` mode](#running-unit-tests) and [isolation techniques](#meteor-specific-isolation) as we use for unit tests.
+
+However, an integration test that crosses the client-server boundary of a Meteor application (where the modules under test cross that boundary) requires a different testing infrastructure, namely Meteor's "full app" testing mode. 
+
+Let's take a look at example of both kinds of tests.
+
+<h3 id="simple-integration-test">Simple integration test</h3>
+
+Our resuable components were a natural fit for a unit test; similarly our smart components tend to require an integration test to really be exercised properly, as the job of a smart component is to bring data together and supply it to a reusable component.
+
+In the Todos example app, we have an integration test for the `Lists_show_page` smart component. This test simply ensures that when the correct data is present in the database, the template renders correctly -- that it is gathering the correct data as we expect. It isolates the rendering tree from the more complex data subscription part of the Meteor stack. If we wanted to test that the subscription side of things was working in concert with the smart component, we'd need to write a [full app integration test](#full-app-integration-test).
+
+```js
+/* eslint-env mocha */
+
+import { Meteor } from 'meteor/meteor';
+import { Factory } from 'meteor/factory';
+import { Random } from 'meteor/random';
+import { chai } from 'meteor/practicalmeteor:chai';
+import { StubCollections } from 'meteor/stub-collections';
+import { Template } from 'meteor/templating';
+import { _ } from 'meteor/underscore';
+import { $ } from 'meteor/jquery';
+import { FlowRouter } from 'meteor/kadira:flow-router';
+import { sinon } from 'meteor/practicalmeteor:sinon';
```

Comment:
We have it already as it's implied by `pm:mocha`. Otherwise I'd have used it from NPM